examples/react-app/src/components/counter.tsx

```diff
@@ -141,4 +186,86 @@ export default function ContractCounter({ isSigning, setIsSigning }: Props) {
       console.error(error);
     }
   }
+
+  async function handleSignIncrement() {
+    if (wallet && currentConnector && 'signTransaction' in currentConnector) {
+      setLoading(true);
+      setIsSigning(true);
+      try {
+        const contract = new Counter(counterContractId, wallet);
+
+        const txRequest = await contract.functions
+          .increment_counter()
+          .txParams({ gasLimit: 100_000 })
+          .getTransactionRequest();
+
+        if (!txRequest) {
+          throw Error('Failed to create increment counter transaction request');
+        }
+
+        if (!wallet.provider?.url) {
+          throw new Error(
+            'Provider URL not found to create a local provider for assembleTx',
+          );
+        }
+        const localProvider = new Provider(wallet.provider.url);
+        await localProvider.init();
+
+        const { assembledRequest } =
+          await localProvider.assembleTx<ScriptTransactionRequest>({
+            request: txRequest,
+            feePayerAccount: wallet,
+            accountCoinQuantities: [],
+          });
+
+        if (!assembledRequest) {
+          throw new Error(
+            'Failed to destructure .assembledRequest from localProvider.assembleTx response. The method might not exist or returns a different structure. Check runtime error or logs.',
+          );
+        }
+
+        const signedTransaction = await currentConnector.signTransaction(
+          wallet.address.toString(),
+          assembledRequest,
+        );
```

Comment:
can we use the `useSignTransaction` method, instead of directly in the connector here?